src/vdaf.rs

```diff
@@ -272,43 +247,86 @@ pub struct VerifierMessage<F: FieldElement> {
     pub verifier_share: Verifier<F>,
 
     /// The aggregator's share of the joint randomness, derived from its input share.
+    //
+    // TODO(cjpatton) If `input.joint_rand_len() == 0`, then we don't need to bother with the
+    // joint randomness seed at all and make this optional.
     pub joint_rand_seed_share: Key,
 }
 
-/// The initial state of an aggregate upon receiving the [`UploadMessage`] message from the client.
+/// The state of each aggregator.
 #[derive(Clone, Debug)]
-pub struct AggregatorState<V: Value> {
-    phantom: PhantomData<V::Field>,
-    input_share: V,
-    joint_rand_seed: Key,
+pub enum AggregatorState<V: Value> {
+    /// Aggregator is waiting for its input share.
+    Ready {
+        /// Aggregator ID.
+        aggregator_id: u8,
+
+        /// Input parameter, needed to reconstruct a [`Value`] from a vector of field elements.
+        input_param: V::Param,
```

Comment:
Does this need to be carried around in this state object? Isn't it always re-derivable from the PPM task definition?

Reply:
Yes,  in theory it's one of the task parameters. From a code standpoint It's required for verify_start(), and I thought it would be cleanest to pass it through the initial state rather than pass it as a function parameter there. Purely a matter of taste.

Reply:
It's likely that the opaque `helper_state` blob in a PPM `AggregateResp` will contain a serialized `struct AggregatorState`, so besides code esthetics, including this means bigger messages exchanged between the aggregators and more storage required by the leader. I'm happy to move forward with this and optimize message sizes later, though.